### PR TITLE
elements: add 'confidential_assets' & 'confidential_address_prefix' to JSON files

### PR DESCRIPTION
**Pull request by romanz**
Enable `confidential_assets` for Elements (following #66 and #317) to allow
[deriving confidential addresses](https://github.com/romanz/trezor-firmware/blob/a9ec7775b4fe01a0ab6814970f3ba1189dffa002/core/src/apps/wallet/sign_tx/addresses.py#L130) and [signing confidential transactions](https://github.com/romanz/trezor-firmware/blob/a9ec7775b4fe01a0ab6814970f3ba1189dffa002/core/src/apps/wallet/sign_tx/signing.py#L521-L523).

The following Python helper script was used to update the JSON files:
```
import json
import sys

for f in sys.argv[1:]:
    d = json.load(open(f))
    d["confidential_assets"] = False
    d["confidential_address_prefix"] = 0
    with open(f, "w") as o:
        json.dump(d, o, indent=2)
        o.write("\n")
```

Also, set `confidential_address_prefix=4` for Elements.

`coins.json` and `coininfo.py` were re-generated using:
```
$ pipenv run make gen gen_check
```

### CODE REVIEW

File: common/defs/bitcoin/faircoin.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "coin_label": "FairCoin",
   "website": "https://www.faircoin.world",
   "github": "https://github.com/faircoin/faircoin",
-  "maintainer": "Santi Noreña <santi@criptomart.net>",
+  "maintainer": "Santi Nore\u00f1a <santi@criptomart.net>",
```
Comment (prusnak) [inline common/defs/bitcoin/faircoin.json]: Please, let's not change this.

Comment (romanz) [reply common/defs/bitcoin/faircoin.json]: Oops, sorry for the mistake... 
Fixed and force-pushed https://github.com/trezor/trezor-firmware/pull/399/commits/1528853c4de0c964294c74d28919e4a2b4c4f3bf.

**Comment by prusnak** [comment]
@romanz Will there ever be a coin where `confidential_assets` is `True` but `confidential_address_prefix` is `0`? Or the other way around? I am thinking we might drop the `confidential_assets` field and just keep the prefix one when we it is always true that `confidential_assets == confidential_address_prefix > 0`. What do you think?

**Comment by romanz** [comment]
Definitely, it could work, but we'll probably also need to add a separate "blech32" prefix (for confidential native segwit addresses), which would be different from the "bech32" prefix (for explicit native segwit addresses):
https://github.com/ElementsProject/elements/blob/c67ef2938c7f59ff263c13efe5f87fd0f835c34c/src/chainparams.cpp#L491-L492

Maybe we can store the confidential assets'-related information as an "optional" dictionary in the JSON, so most of the coins will leave it as `None`, but in Elements this dictionary will be used to populate the needed `confidential_assets` attribute of `CoinInfo` (e.g. https://github.com/romanz/trezor-firmware/commit/2ce332d419f19b0cf55c3594134e0833e23b861b). 
What do you think?

**Comment by prusnak** [comment]
Yes, 2ce332d makes more sense. Please force push that commit into this branch and I will merge this.

**Comment by romanz** [comment]
> Yes, [2ce332d](https://github.com/trezor/trezor-firmware/commit/2ce332d419f19b0cf55c3594134e0833e23b861b) makes more sense. Please force push that commit into this branch and I will merge this.

Thanks, force-pushed :)

**Comment by prusnak** [comment]
Thanks!

**Pull request merged**